docs/driverapi.md

```diff
@@ -0,0 +1,398 @@
+# Driver API
+
+Below is a pictorial representation of the `Driver API`:
+
+![driverapi.png](imgs/driverapi.png)
+
+## Serialization
+
+The Driver API recommends to use the `protobuf` protocol to serialize public parameters, token requests.
```

Comment:
It recommends `protobuf` but does it support other alternatives? should we name them?